src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/HMPPSAuthService.kt

```diff
@@ -45,6 +45,19 @@ class HMPPSAuthService(
   @Value("\${hmppsauth.api.locations.user-detail}") private val userDetailLocation: String,
   private val hmppsAuthApiWebClient: WebClient,
 ) {
+  fun getUserGroups(user: AuthUser): List<AuthGroupID>? {
+    val url = UriComponentsBuilder.fromPath(authUserGroupsLocation)
+      .buildAndExpand(user.userName)
+      .toString()
+
+    return hmppsAuthApiWebClient.get().uri(url)
+      .retrieve()
+      .onStatus({ HttpStatus.NOT_FOUND == it }, { Mono.just(null) })
```

Comment:
what do you think of this method returning a nullable list vs throwing an error?